presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java

```diff
@@ -109,32 +109,32 @@
     private final boolean clientTransactionSupport;
     private final String clientInfo;
 
-    public HttpRequestSessionContext(HeaderSupport forwardedHeaderSupport, HttpServletRequest servletRequest)
+    public HttpRequestSessionContext(HeaderSupport forwardedHeaderSupport, HttpHeaders headers, String remoteAddress, Optional<Identity> authenticatedIdentity)
             throws WebApplicationException
     {
-        catalog = trimEmptyToNull(servletRequest.getHeader(PRESTO_CATALOG));
-        schema = trimEmptyToNull(servletRequest.getHeader(PRESTO_SCHEMA));
-        path = trimEmptyToNull(servletRequest.getHeader(PRESTO_PATH));
+        catalog = trimEmptyToNull(headers.getHeaderString(PRESTO_CATALOG));
```

Comment:
I don't think this method does what you think:
```
     * @return the HTTP header value. If the HTTP header is not present then
     *         {@code null} is returned. If the HTTP header is present but has no
     *         value then the empty string is returned. If the HTTP header is present
     *         more than once then the values of joined together and separated by a ','
     *         character.
```
I think you want `headers.getRequestHeaders().getFirst(name)`

Reply:
Maybe we should have a helper method that fails if the header is specified multiple times.

Reply:
That would be a backwards incompatible change since the current code does "If there are multiple headers with the same name, this method returns the first head in the request."